api/v1beta1/metal3machine_types.go

```diff
@@ -91,6 +93,19 @@ func (s *Metal3MachineSpec) IsValid() error {
 	if len(missing) > 0 {
 		return errors.Errorf("Missing fields from ProviderSpec: %v", missing)
 	}
+
+	invalid := []string{}
+	_, err := url.ParseRequestURI(s.Image.URL)
+	if err != nil {
+		invalid = append(invalid, "Image.URL")
+	}
+	_, err = url.ParseRequestURI(s.Image.Checksum)
+	if err != nil {
+		invalid = append(invalid, "Image.Checksum")
+	}
+	if len(invalid) > 0 {
+		return errors.Errorf("Invalid URL: %v", invalid)
```

Comment:
Is this log accurate considering the case where there is an invalid CheckSum as well?

Reply:
Because the above verifies whether the URL and Checksum are in the correct url format, the URL here is the representation format instead of Image.URL.
Maybe my log information is not well expressed.
What do you think about this change:
```
return errors.Errorf("The following fields are invalid URLs: %v", invalid)
```

Reply:
To keep it consistent with the previous validation (missing), it could be:
```
return errors.Errorf("Invalid fields from ProviderSpec: %v", invalid)
```

Reply:
updated, thanks.